src/plots/layout_attributes.js

```diff
@@ -444,6 +445,112 @@ module.exports = {
+            width: {
+                valType: 'number',
+                min: 0,
+                dflt: 4,
+                role: 'info',
+                editType: 'none',
+                description: 'Sets the line width (in px).'
+            },
+            dash: extendFlat({}, dash, {
+                dflt: 'solid',
+                editType: 'none'
+            }),
+            role: 'info',
+            editType: 'none'
+        },
+        fillcolor: {
+            valType: 'color',
+            dflt: 'rgba(0,0,0,0)',
+            role: 'info',
+            editType: 'none',
+            description: 'Sets the color filling new shapes\' interior.'
+        },
+        fillrule: {
+            valType: 'enumerated',
+            values: ['evenodd', 'nonzero'],
+            dflt: 'evenodd',
+            role: 'info',
+            editType: 'none',
+            description: [
+                'Determines the path\'s interior.',
+                'For more info please visit https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill-rule'
+            ].join(' ')
+        },
+        opacity: {
+            valType: 'number',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            role: 'info',
+            editType: 'none',
+            description: 'Sets the opacity of new shapes.'
+        },
+        layer: {
+            valType: 'enumerated',
+            values: ['below', 'above'],
+            dflt: 'above',
+            role: 'info',
+            editType: 'none',
+            description: 'Specifies whether new shapes are drawn below or above traces.'
+        },
+        drawdirection: {
+            valType: 'enumerated',
+            role: 'info',
+            values: ['ortho', 'horizontal', 'vertical', 'diagonal'],
+            dflt: 'diagonal',
+            editType: 'none',
+            description: [
+                'When `dragmode` is set to *drawrect*, *drawline* or *drawcircle*',
+                'this limits the drag to be horizontal, vertical or diagonal.',
+                'Using *diagonal* there is no limit e.g. in drawing lines in any direction.',
+                '*ortho* limits the draw to be either horizontal or vertical.',
+                '*horizontal* allows horizontal extend.',
+                '*vertical* allows vertical extend.'
+            ].join(' ')
+        },
+
+        editType: 'none'
+    },
+
+    activeshape: {
+        fillcolor: {
+            valType: 'color',
+            dflt: 'rgb(255,0,255)',
+            role: 'style',
+            editType: 'none',
+            description: 'Sets the color filling the active shape\' interior.'
+        },
+        opacity: {
+            valType: 'number',
+            min: 0,
+            max: 1,
+            dflt: 0.5,
+            role: 'info',
+            editType: 'none',
+            description: [
+                'Sets the opacity of the active shape.',
+                'If using a value greater than half,',
```

Comment:
this doesn't seem accurate... It looks like it's more on the other side: if an **inactive** shape has an opacity greater than 0.5, then we can't create a new shape by starting a click-drag inside that shape.

Reply:
Good eye. Fixed in https://github.com/plotly/plotly.js/pull/4775/commits/cedad524b52d2b373f3d801f9ce354804f0e3c0a.